web3.md

```diff
@@ -1018,5 +1020,47 @@ loadCallSettings
  // -------------------------------------
     rule <k> #eth_call_finalize => #popNetworkState ~> #sendResponse ("result": #unparseDataByteArray( OUTPUT )) ... </k>
          <output> OUTPUT </output>
-endmodule
 ```
+
+- `#eth_estimateGas`
+**TODO**: avoid EVMC_OUT_OF_GAS
+**TODO**: implement funcionality for block number argument
+
+```k
+    syntax KItem ::= "#eth_estimateGas"
+ // -----------------------------------
+    rule <k> #eth_estimateGas
+          => #pushNetworkState
+          ~> mkTX !ID:Int
+          ~> #loadNonce #parseHexWord(#getString("from", J)) !ID
+          ~> load "transaction" : { !ID : J }
+          ~> signTX !ID #parseHexWord(#getString("from", J))
+          ~> #setMode ACCUMULATE_GAS
+          ~> #prepareTx !ID
+          ~> #eth_estimateGas_finalize GUSED
+         ...
+         </k>
+         <params> [ ({ _ } #as J), TAG, .JSONList ] </params>
+         <gasUsed> GUSED </gasUsed>
+      requires isString(#getJSON("from", J) )
```

Comment:
Load the `<gasLimit>` cell into the `<gas>` cell, and then if `J` has the `gasLimit` field, it will overwrite it.